src/main/java/org/jenkinsci/plugins/workflow/steps/EnvStep.java

```diff
@@ -178,10 +178,7 @@
 
         private static List<String> toKeyValueList(Map<?, ?> map) {
             return map.entrySet().stream()
-                .filter(e -> e.getKey() instanceof String)
-                .collect(Collectors.toMap(e -> (String)e.getKey(), e -> e.getValue() == null ? "" : e.getValue().toString()))
-                .entrySet().stream()
-                .map(m -> m.getKey() + "=" + m.getValue())
+                .map(m -> m.getKey() + "=" + (m.getValue() == null ? "" : m.getValue().toString()))
```

Comment:
```suggestion
                .map(m -> (String) m.getKey() + "=" + (m.getValue() == null ? "" : m.getValue().toString()))
```

since non-`String` keys would be bogus

Reply:
IDE consider this a redundant cast and wants to remove it.

Reply:
Then your IDE is wrong and you should ignore it. You should be able to have a test demonstrating a failure from some bogus thing like

```groovy
withEnv([new Object(): 'x']) {}
```

Reply:
This case is not supported:
```
WorkflowScript: 2: illegal colon after argument expression;
  solution: a complex label expression before a colon must be parenthesized @ line 2, column 72.
  D: true, E: null, new Object(): 'x']) {
```

Reply:
Maybe

```groovy
withEnv([(new Object()): 'x']) {}
```

then, whatever the syntax is. At worst

```groovy
def m = [:]
m.put(new Object(), 'x')
withEnv(m) {}
```

Reply:
Than you get into the whole script security: one you add `(new Object())` `org.jenkinsci.plugins.scriptsecurity.sandbox.RejectedAccessException: Scripts not permitted to use new java.lang.Object`

Reply:
I'd argue your asking for trouble

Reply:
OK forget `new Object()`, just some arbitrary object key which is _not_ a `String`. `new URL('http://nowhere.net/')` for example is whitelisted.

Reply:
Than you get into `java.lang.ClassCastException: java.net.URL cannot be cast to java.lang.String` 🤔 

Properly better to call `toString` instead of trying to cast the object

Reply:
> Then you get into `java.lang.ClassCastException`

Exactly what I was requesting with 8d385a2.

Reply:
![image](https://user-images.githubusercontent.com/1661688/74883319-0051eb80-5371-11ea-9f7a-3b543cc8e750.png)